### PR TITLE
UI refactor, added webpack and React

### PR DESCRIPTION
**Pull request by dangerousfood**

### CODE REVIEW

**Comment by dangerousfood** [comment]
Build size should be 300kb

**Comment by jrhea** [comment]
interesting additions to your. GIT ignore... 👀

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,4 @@
+.DS_Store
+node_modules
+homework/anime_porn
```
Comment (jrhea) [inline .gitignore]: 👍

File: .gitmodules [hunk 2]
```diff
@@ -0,0 +1,3 @@
+[submodule "solr-js"]
```
Comment (jrhea) [inline .gitmodules]: Next iteration i'd like to get rid of the submodule and just add the logic of pulling down the solr-js repo as part of the build.  We could put all the logic for "building for prod" in a makefile even

**Pull request merged**